build-tools/src/main/java/co/elastic/apm/compile/tools/publishing/ApmPublisherRootPlugin.java

```diff
@@ -35,12 +37,29 @@ public void apply(Project project) {
     private void configureVersion(Project project) {
         String versionOverride = getVersionOverride(project);
         if (versionOverride != null) {
+            validateVersionOverrideFormatting(versionOverride);
+            validateVersionOverrideValues(project.getVersion().toString(), versionOverride);
             System.out.println("Overriding version with: '" + versionOverride + "'");
             project.setVersion(versionOverride);
             project.subprojects(subproject -> subproject.setVersion(versionOverride));
         }
     }
 
+    private static void validateVersionOverrideValues(String currentVersion, String versionOverride) {
```

Comment:
IIUC, this function validates that only `patches` are allowed. If so, could the function's name reflect that? Or do you envision to support some other cases in the future?

Reply:
Got it, I see what you mean. I had an idea of allowing also lower numbers in case we wanted to release a patch version for a very old one, but now that I think about it better, I guess in that case we'd create a branch off of the version tag which will have the old version set in `gradle.properties` file already, essentially just making this parameter to still validate only patch number changes are allowed anyway. I'll update it to make it clearer, cheers!

Reply:
It should be better now 👍